### PR TITLE
Aesthetic and UI/UX enhancements

### PR DESCRIPTION
**Pull request by AJun01**
I am creating the first Pull Request to get familiar with it.

Two commits: The first week included: fixed some typos in READMEs, and installed some dependencies(nodemon) in order to fix the server app crash. 

Two commits:The second week changes included: fixed desktop-app main process function error, fixed a UI inconsistent issue. 

The first PR of my career ;)

### CODE REVIEW

**Comment by iskitsas** [comment]
@AJun01 The title is misleading, I would expect to review code changes related with a UI bug (I suppose for the desktop-app). However in the description, you are describing that you have resolved a crash for the server-app.

Include in your description the issues you are resolving. Create separate ones if needed, and tag them with `gsoc24` label.

Some comments/questions:
- Is this PR related with #68?
- I see two commits: 6a2e82b and cf65222. Do they fix a problem you have identified? Create a github issue with more details on what exactly the problem was, how you have identified it and so on

**Comment by AJun01** [comment]
Sorry for the confusion, I am still familiarizing myself with Git so I may make some mistakes. I included a couple of things in this one PR, which is really misleading, I will edit this into the first version only for fixed typos and server-app crashes. I will create another PR to include the content of the second week: Desktop-app error and a minor UI fix.

**Comment by AJun01** [comment]
> @AJun01 The title is misleading, I would expect to review code changes related with a UI bug (I suppose for the desktop-app). However in the description, you are describing that you have resolved a crash for the server-app.
> 
> Include in your description the issues you are resolving. Create separate ones if needed, and tag them with `gsoc24` label.
> 
> Some comments/questions:
> 
> * Is this PR related with [UI Inconsistent Fix: The animation bar in the simulation connection appears too short  #68](https://github.com/flexivian/flexbench/issues/68)?
> * I see two commits: [6a2e82b](https://github.com/flexivian/flexbench/commit/6a2e82b6d5881265f75d82e86e195619f3e5b6b2) and [cf65222](https://github.com/flexivian/flexbench/commit/cf65222122dcded7b876499a3a341461825fd26b). Do they fix a problem you have identified? Create a github issue with more details on what exactly the problem was, how you have identified it and so on

**Comment by AJun01** [comment]
> Sorry for the confusion, I am still familiarizing myself with Git so I may make some mistakes. I included a couple of things in this one PR, which is misleading, I will edit this into the first version only for fixed typos and server-app crashes. I will create another PR to include the content of the second week: Desktop-app error and a minor UI fix.

I have tried to separate a new PR by creating a new branch, but soon I realized that the codebase is the same because I have already committed and pushed the changes to my repo and this PR automatically updated with the newest changes. So, this PR will include four different committed and pushed changes from these two weeks.

**Comment by iskitsas** [comment]
@AJun01 can you include some before/after screenshots of the UI inconsistency issue you have fixed?

**Comment by AJun01** [comment]
before: 
image
after: 
image
image
image

**Comment by AJun01** [comment]
but I noticed that the PC logo is no at the center of the circle, I will fix it ASAP

File: compose-dev.yaml
Comment (mariosk) [inline compose-dev.yaml:1]: @AJun01 what is this file used for? If it's a docker-compose file why it is located in this folder?

**Comment by mariosk** [comment]
> @AJun01 The title is misleading, I would expect to review code changes related with a UI bug (I suppose for the desktop-app). However in the description, you are describing that you have resolved a crash for the server-app.
> 
> Include in your description the issues you are resolving. Create separate ones if needed, and tag them with `gsoc24` label.
> 
> Some comments/questions:
> 
> * Is this PR related with [UI Inconsistent Fix: The animation bar in the simulation connection appears too short  #68](https://github.com/flexivian/flexbench/issues/68)?
> * I see two commits: [6a2e82b](https://github.com/flexivian/flexbench/commit/6a2e82b6d5881265f75d82e86e195619f3e5b6b2) and [cf65222](https://github.com/flexivian/flexbench/commit/cf65222122dcded7b876499a3a341461825fd26b). Do they fix a problem you have identified? Create a github issue with more details on what exactly the problem was, how you have identified it and so on

I totally agree with @iskitsas on the above comments. My two cents @AJun01:

Please create descriptive Github issues (bugs) with logs and screenshots on what is the exact issue you found.
Then, when you create a branch to resolve an issue, the branch name should be like this:
$USER/fix-issue-##

Here is an example:
**mariosk/fix-issue-73 where 73 is this one: https://github.com/flexivian/flexbench/issues/73**

Now regarding the commit messages of your commits, you should use the semantics of commit messages:

**Type**: Indicates the type of change. Common types include:

- feat: A new feature
- fix: A bug fix
- docs: Documentation only changes
- style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- refactor: A code change that neither fixes a bug nor adds a feature
- perf: A code change that improves performance
- test: Adding missing or correcting existing tests
- chore: Changes to the build process or auxiliary tools and libraries such as documentation generation

**Scope**: Indicates the section of the codebase this commit affects (optional but recommended).

**Subject**: A brief description of the change.

**Body**: A more detailed explanation of the changes (optional).

**Footer**: Information about breaking changes or issues this commit closes (optional).

Here is an example:
```
feat(authentication): add JWT authentication

- Added JWT token generation during user login
- Updated user model to include token expiration
- Created middleware to validate JWT on protected routes
- Modified login endpoint to return JWT on successful login

BREAKING CHANGE: Users need to update their login process to handle JWT tokens
```

In this example:

The type is **feat** (indicating a new feature).
The scope is **authentication**.
The subject is **add JWT authentication**.
The body provides detailed information about what was changed and why.
The footer mentions a **BREAKING CHANGE** indicating that users need to update their login process.

File: compose-dev.yaml
Comment (AJun01) [reply compose-dev.yaml:1]: I am not sure about this one but maybe it was an automatic change made by Docker.

**Comment by AJun01** [comment]
> > @AJun01 The title is misleading, I would expect to review code changes related with a UI bug (I suppose for the desktop-app). However in the description, you are describing that you have resolved a crash for the server-app.
> > Include in your description the issues you are resolving. Create separate ones if needed, and tag them with `gsoc24` label.
> > Some comments/questions:
> > 
> > * Is this PR related with [UI Inconsistent Fix: The animation bar in the simulation connection appears too short  #68](https://github.com/flexivian/flexbench/issues/68)?
> > * I see two commits: [6a2e82b](https://github.com/flexivian/flexbench/commit/6a2e82b6d5881265f75d82e86e195619f3e5b6b2) and [cf65222](https://github.com/flexivian/flexbench/commit/cf65222122dcded7b876499a3a341461825fd26b). Do they fix a problem you have identified? Create a github issue with more details on what exactly the problem was, how you have identified it and so on
> 
> I totally agree with @iskitsas on the above comments. My two cents @AJun01:
> 
> Please create descriptive Github issues (bugs) with logs and screenshots on what is the exact issue you found. Then, when you create a branch to resolve an issue, the branch name should be like this: $USER/fix-issue-##
> 
> Here is an example: **mariosk/fix-issue-73 where 73 is this one: #73**
> 
> Now regarding the commit messages of your commits, you should use the semantics of commit messages:
> 
> **Type**: Indicates the type of change. Common types include:
> 
> * feat: A new feature
> * fix: A bug fix
> * docs: Documentation only changes
> * style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
> * refactor: A code change that neither fixes a bug nor adds a feature
> * perf: A code change that improves performance
> * test: Adding missing or correcting existing tests
> * chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
> 
> **Scope**: Indicates the section of the codebase this commit affects (optional but recommended).
> 
> **Subject**: A brief description of the change.
> 
> **Body**: A more detailed explanation of the changes (optional).
> 
> **Footer**: Information about breaking changes or issues this commit closes (optional).
> 
> Here is an example:
> 
> ```
> feat(authentication): add JWT authentication
> 
> - Added JWT token generation during user login
> - Updated user model to include token expiration
> - Created middleware to validate JWT on protected routes
> - Modified login endpoint to return JWT on successful login
> 
> BREAKING CHANGE: Users need to update their login process to handle JWT tokens
> ```
> 
> In this example:
> 
> The type is **feat** (indicating a new feature). The scope is **authentication**. The subject is **add JWT authentication**. The body provides detailed information about what was changed and why. The footer mentions a **BREAKING CHANGE** indicating that users need to update their login process.

Thank you ! I will follow the example and template later on!

File: compose-dev.yaml
Comment (mariosk) [reply compose-dev.yaml:1]: ok no worries, in general if you see files that you did not add, you may remove them.

**Comment by mariosk** [comment]
@iskitsas following our offline discussion I'm adding here the screenshot that maybe the evidence that something has broken in the UI:

image

It seems that the monitor is not centered any more.

**Comment by AJun01** [comment]
> @iskitsas following our offline discussion I'm adding here the screenshot that maybe the evidence that something has broken in the UI:
> 
> image
> 
> It seems that the monitor is not centered any more.

Yes, I have noticed earlier, fixed with my last commit. thank you for point out!
image

**Comment by AJun01** [comment]
image

**Comment by AJun01** [comment]
Closing this PR to open a new one with the renamed branch.

**Comment by AJun01** [comment]
Closing this PR to open a new one with the renamed branch

**Pull request closed without merging**